src/main/java/net/minecraftforge/event/world/DimensionPreloadEvent.java

```diff
@@ -0,0 +1,33 @@
+package net.minecraftforge.event.world;
+
+import net.minecraft.world.DimensionType;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event;
+import net.minecraftforge.common.MinecraftForge;
+
+/**
+ * This event is fired when DimensionManager is starting to init a new dimension
+ * with {@link DimensionManager#initDimension(int)},
+ * <br>
+ * This event is {@link Cancelable}.<br>
+ * <br>
+ * event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
+ * special care should be used when canceling this event
+ * use it only when you know what you are doing
```

Comment:
I cannot give an example of my own yet as i don't have my plugin/mod ready yet but i will give more details on how the FE does this event canceling. On line 418 in https://github.com/ForgeEssentials/ForgeEssentials/blob/110/develop/src/main/java/com/forgeessentials/multiworld/MultiworldManager.java you can see exactly how an event like this can be used for this purpose. I just want to prevent having to use "mixin" when this can be achieved by having a simple event in the right place in forge instead. The point of an API is to make coding easier so this is what i try to help with, by giving a simpler solution to this specific problem.